config/peerpods/podvm/aws-podvm-image-handler.sh

```diff
@@ -287,10 +312,106 @@ function delete_ami_id_annotation_from_peer_pods_cm() {
     echo "Ami id annotation deleted from peer-pods-cm configmap successfully"
 }
 
-# Function to create the ami in AWS
+function create_ami_from_prebuilt_artifact() {
+    echo "Creating AWS AMI image from prebuilt artifact"
 
-function create_ami() {
-    echo "Creating AWS AMI"
+    echo "Pulling the podvm image from the provided path"
+    image_src="/tmp/image"
+    extraction_destination_path="/image"
+    image_repo_auth_file="/tmp/regauth/auth.json"
+
+    [[ ! ${BUCKET_NAME} ]] && error_exit "BUCKET_NAME is not defined"
+
+    # Get the PODVM_IMAGE_TYPE, PODVM_IMAGE_TAG and PODVM_IMAGE_SRC_PATH
+    get_image_type_url_and_path
+
+    case "${PODVM_IMAGE_TYPE}" in
+    oci) # TODO: test
+        echo "Extracting the raw image from the given path."
+
+        mkdir -p "${extraction_destination_path}" ||
+            error_exit "Failed to create the image directory"
+
+        extract_container_image "${PODVM_IMAGE_URL}" \
+            "${PODVM_IMAGE_TAG}" \
+            "${image_src}" \
+            "${extraction_destination_path}" \
+            "${image_repo_auth_file}"
+
+        # Form the path of the podvm vhd image.
+        podvm_image_path="${extraction_destination_path}/rootfs/${PODVM_IMAGE_SRC_PATH}"
+
+        upload_image_to_s3
```

Comment:
Maybe I'm missing something here too. But with Azure, you are uploading the disk and creating image outside the bootc) and oci) case. IIUC, for the bootc case here, the image would not be uploaded.

Did I got this wrong?

Reply:
bootc image builder supports native upload for AWS, hence i used it, other cloud providers are expected to be supported in the future as well.
I can implement the upload myself for AWS, however, i believe it will increase the points of failure

Reply:
I think splitting it would help keep things consistent, but I'll leave that decision for this PR up to you, we might split it later.

Said that, if we keep it as-is, the function name could be clearer since `bootc_to_ami` sounds like it just creates the AMI.

Reply:
@snir911 ,

don't you need to convert qcow -> raw before `upload_image_to_s3`?

Reply:
Looks like bootc-image-builder either allows you to create qcow2/raw or directly create AMI. There is no intermediate step of using a qcow2 or raw and create ami. At least I couldn't find it in the documents.

Reply:
The block where `upload_image_to_s3` is called will handle the oci case.